core/src/main/kotlin/org/evomaster/core/problem/rest/service/AbstractRestSampler.kt

```diff
@@ -80,13 +84,43 @@ abstract class AbstractRestSampler : HttpWsSampler<RestIndividual>() {
         val openApiSchema = problem.openApiSchema
 
         if(!openApiURL.isNullOrBlank()) {
-            swagger = OpenApiAccess.getOpenAPIFromURL(openApiURL)
+
+            // first try to retrieve the OpenAPI without authentication
+            swagger = OpenApiAccess.getOpenAPIFromURL(openApiURL, HttpWsNoAuth())
+
+            // if it fails, try to get swagger using each authentication information.
+            // if swagger failed due to unauthorized access, then an empty swagger is retrieved
+            if(swagger.paths == null) {
```

Comment:
this will conflate the case in which the schema is read properly, but empty... 
rather should have thrown exception, and here have a try/catch for that type (where error message in a constant, and,then, if different, re-throw the exception)

Reply:
Hi Andrea,

Thanks for your comment. I have fixed this one. Now I just throw an exception if the swagger cannot be retrieved.